### ISSUE TITLE
Incorrect name ambiguity with discard lambda parameters

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#61143]
**Version Used**: 
```
commit 18305a1f4b82de5092cc8f90573afa5a6d091cf8
Merge: afe689ab1a9 265022a3e73
Author: Manish Vasani <mavasani@microsoft.com>
Date:   Fri Apr 29 09:21:23 2022 -0700

    Merge pull request #60906 from mavasani/FixAllCodeRefactoring_2

    Add FixAll support for code refactorings (approach #2)
```

**Steps to Reproduce**:

Compile the following code with preview language version.
```cs
class AAttribute : System.Attribute
{
  public AAttribute(string s) { }
}

class C
{
  void M(int _)
  {
    System.Func<string, string, int> a = [A(nameof(_))](_, _) => _;
  }
}
```

**Expected Behavior**:
The code compiles. `nameof(_)` references to the parameter `int _` since lambda parameters are discards and not actual elements.
Note that `_` in the lambda body correctly binds to `int _` and doesn't consider discard elements

**Actual Behavior**:
Incorrect and strangely worded error is reported:
`error CS0229: Ambiguity between 'string' and 'string'`

**Notes**:
Note that this is also a breaking change as the same code compiles with C# 10. It's a small one and in line with other breaking changes caused by parameter scope changes, but given that here it doesn't bring any new functionality it seems avoidable.
